### PR TITLE
Feature/express 4 migration

### LINKED ISSUE #140: Migrate all Express code to use version 4

### PR DESCRIPTION
**Pull request by Engineering-Robert**
## Issue

Fixes #140 

## Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Now using Express 4.x instead of version 3.

### How Has This Been Tested?

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

---
This change is [Reviewable](https://reviewable.io/reviews/vinitkumar/node-twitter/181)

### CODE REVIEW

**Comment by Engineering-Robert** [comment]
@vinitkumar I'll let you merge this into master.

**Comment by vinitkumar** [comment]
@Engineering-Robert Mostly looks great! One question though, did we needed to move server.js into app? I intentionally kept it outside as it has little to do with app code.

File: config/express.js [hunk 1]
```diff
@@ -17,63 +24,58 @@ module.exports = (app, config, passport) => {
...
 
-      res.status(500).render("pages/500", { error: err.stack });
-    });
+  app.use((err, req, res, next) => {
+    if (err.message.indexOf("not found") !== -1) {
+      return next();
+    }
+    console.log(err.stack);
 
-    app.use((req, res) => {
```
Comment (vinitkumar) [inline config/express.js]: Any reason to remove the render of 404 page on not found routes?

**Comment by Engineering-Robert** [comment]
A lot of projects i've been looking at have their main server file in a "src" folder so I wanted to follow along with that practice by adding our server.js into the app folder. This way our root directory only contains dot files, READMEs, and other config stuff. Thoughts?

File: config/express.js (hunk 1 above)
Comment (Engineering-Robert) [reply config/express.js]: Ya, the 404 page was always being rendered when it was included in express.js, so I had to move it to the bottom of routes.js. I think it has to do with not using app.router (deprecated) anymore.

Comment (vinitkumar) [reply config/express.js]: Okay!

**Pull request merged**